ssl/statem/extensions_clnt.c

```diff
@@ -678,12 +678,41 @@ EXT_RETURN tls_construct_ctos_early_data(SSL *s, WPACKET *pkt,
                                          unsigned int context, X509 *x,
                                          size_t chainidx, int *al)
 {
+    const unsigned char *id;
+    size_t idlen;
+    SSL_SESSION *psksess = NULL;
+    const EVP_MD *handmd = NULL;
+
+    if (s->hello_retry_request)
+        handmd = ssl_handshake_md(s);
```

Comment:
> Early data is not permitted after HelloRetryRequest.

Shouldn't we just return early here?
I guess that's inconsistent with moving the psk_use_session_cb() here from construct_psk but we still need to disable early data for HelloRetryRequest resposes, which I don't see happening here (am I missing it?).  We certainly can't rely on the callback detecting this case...